tests/test_proxy_functional.py

```diff
@@ -3,13 +3,52 @@
 import pathlib
 from unittest import mock
 
+import proxy
 import pytest
+import trustme
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
 
+@pytest.fixture
+def trustme_certs(tmp_path):
+    ca = trustme.CA()
+    ca.issue_cert("localhost")
+    ca_cert_path = tmp_path / "ca.pem"
+    ca.cert_pem.write_to_path(ca_cert_path)
+    ca_key_path = tmp_path / "server.key"
+    ca.private_key_pem.write_to_path(ca_key_path)
+    return (str(ca_cert_path), str(ca_key_path))
+
+
+@pytest.fixture
+def secure_proxy_url(trustme_certs):
+    port = proxy.common.utils.get_available_port()
+    input_args = [
+        "--num-workers",
+        "1",
+        "--threadless",
+        "--hostname",
+        "0.0.0.0",
+        "--port",
+        str(port),
+        "--cert-file",
+        trustme_certs[0],
+        "--key-file",
+        trustme_certs[1],
+    ]
+    proxy_url = "http://0.0.0.0:" + str(port)
+    with proxy.Proxy(input_args=input_args) as proxy_instance:
```

Comment:
1) If you don't use `proxy_instance`, no need to declare it
2) Does `proxy_instance` have any attribute exposing its real port? It'd be useful for ephemeral port binds

Reply:
I'll see if I can find one. I looked briefly but didn't see it. I'll look again.

Reply:
I looked into this again, but I didn't see a way to do it. When I interactively debug [here](https://github.com/abhinavsingh/proxy.py/blob/develop/proxy/core/acceptor/pool.py#L110) I can see the port, but when I access the `proxy_instance.acceptors.socket` it's closed...  Unless you have a pointer I'm going to leave it at `proxy.common.utils.get_available_port()` which I also don't like, but it's what proxy.py's code itself also does. Any ideas are welcome.

The only other way I know is to call to netstat and parse the response, but that seems unreliable.

Reply:
Well, at least use our own fixture for this.